src/main/java/seedu/address/logic/commands/AddProjectCommand.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.project.Project;
+
+/**
+ * Adds a project to the CoLAB.
+ */
+public class AddProjectCommand extends Command {
+
+    public static final String COMMAND_WORD = "addP";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a project to CoLAB. "
+            + "Parameters: "
+            + PREFIX_NAME + "PROJECT_NAME";
+
+    public static final String MESSAGE_SUCCESS = "New project added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PROJECT = "This project already exists";
+
+    private final Project toAdd;
+
+    /**
+     * Creates an AddProjectCommand to add the specified {@code Project}
+     */
+    public AddProjectCommand(Project project) {
+        requireNonNull(project);
+        toAdd = project;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (model.hasProject(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PROJECT);
+        }
+
+        model.addProject(toAdd);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
```

Comment:
wah this is complicated, do you want to split it one by one before return

Reply:
This was copied over from the AddCommand. I agree with you, maybe we should open an issue and fix all of them at once?